integration/daemon/daemon_test.go

```diff
@@ -308,43 +327,34 @@ func TestDaemonProxy(t *testing.T) {
 
 		err := d.StartWithError("--http-proxy", proxyRawURL, "--https-proxy", proxyRawURL, "--no-proxy", "example.com", "--config-file", configFile, "--validate")
 		assert.ErrorContains(t, err, "daemon exited during startup")
-		logs, err := d.ReadLogFile()
-		assert.NilError(t, err)
+
 		expected := fmt.Sprintf(
 			`the following directives are specified both as a flag and in the configuration file: http-proxy: (from flag: %[1]s, from file: %[1]s), https-proxy: (from flag: %[1]s, from file: %[1]s), no-proxy: (from flag: example.com, from file: example.com)`,
 			proxyURL,
 		)
-		assert.Assert(t, is.Contains(string(logs), expected))
+		poll.WaitOn(t, d.PollCheckLogs(ctx, daemon.ScanLogsMatchString(expected)))
 	})
 
 	// Make sure values are sanitized when reloading the daemon-config
 	t.Run("reload sanitized", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.TODO()
```

Comment:
I added a single `ctx` here; this should (deliberately) conflict with the `testutil.StartSpan(ctx, t)` in https://github.com/moby/moby/pull/45652, so that we don't forget 😃

Reply:
Why don't we use `context.Background()` here?

Reply:
Because this one needs to be replaced with the otel one 😂 

Either one should be fine I guess